piqueserver/server.py

```diff
@@ -347,15 +348,23 @@ def __init__(self, interface, config):
 
     @inlineCallbacks
     def get_external_ip(self, ip_getter):
+        print('Retrieving external IP from {!r} to generate server identifier.'.format(ip_getter))
         try:
             ip = yield self.getPage(ip_getter)
-        except OSError as e:
+        except Exception as e:
             print("Getting external IP failed:", e)
             return
 
-        self.ip = ip.strip()
+        ip = ip.strip()
+        if not re.match(r'^\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}$', ip):
```

Comment:
I'd prefer just putting this into ip_address to outsource the validation.

Might also be worth making `protocol.ip` an IPV4Address instance... but that might cause backwards-incompatibility. Perhaps something along the lines of renaming it to `ip_address` or something and then making `ip` a property that returns `str(self.ip_address)`? The places where self.ip are used will be very limited though, so I don't think it's at all worth it.

Reply:
:+1:

I don't think protocol.ip is actually used anywhere though.